Sources/TestDRSMacros/Expectations/FunctionCallVerificationMacro.swift

```diff
@@ -0,0 +1,162 @@
+//
+// Created on 1/13/25.
+// Copyright © 2025 Turo Open Source. All rights reserved.
+//
+
+import SwiftCompilerPlugin
+import SwiftDiagnostics
+import SwiftSyntax
+import SwiftSyntaxBuilder
+import SwiftSyntaxMacros
+
+/// Expands macros that verify function calls to transform a reference to the function into a reference and a function signature. The reference to the function must be the first argument in the macro.
+///
+/// For example, the following macro:
+/// ```
+/// #verification(mock.foo(paramOne:), with: "Hello World")
+/// ```
+/// expands to:
+/// ```
+/// verification(
+///     mock.foo(paramOne:),
+///     withSignature: "foo(paramOne:),
+///     expectedInput: "Hello World"
+/// )
+/// ```
+protocol FunctionCallVerificationMacro: ExpressionMacro {
```

Comment:
I renamed `ExpectationMacro` since this expansion code can also be used for the `ConfirmationOfCall` macro.